brave/src/main/java/brave/propagation/TraceContext.java

```diff
@@ -481,7 +488,7 @@ public final TraceContext build() {
       if (spanId == 0L) missing += " spanId";
       if (!"".equals(missing)) throw new IllegalArgumentException("Missing:" + missing);
       return new TraceContext(
-        flags, traceIdHigh, traceId, localRootId, parentId, spanId, ensureImmutable(extra)
+        flags, traceIdHigh, traceId, localRootId, parentId, spanId, ensureImmutable(extraList)
```

Comment:
Sorry my previous comment was invalid I think. I didn't notice the ensure immutable here.

Reply:
there was a bug just not here :)